### PR TITLE
Bug 1740722 Add schema.yaml to urlbar_clients_daily

### PR DESCRIPTION
**Pull request by jklukas**
This will help prevent errors when we add new fields in the future, as seen in https://bugzilla.mozilla.org/show_bug.cgi?id=1740722

Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title)
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated

### CODE REVIEW

**Comment by jklukas** [comment]
cc @linh @kik-kik

File: sql/moz-fx-data-shared-prod/telemetry_derived/urlbar_clients_daily_v1/metadata.yaml [hunk 1]
```diff
@@ -1,12 +1,18 @@
...
   incremental: true
   schedule: daily
 scheduling:
   dag_name: bqetl_urlbar
+bigquery:
+  time_partitioning:
+    field: submission_date
+    type: day
+    require_partition_filter: true
+  clustering: null
```
Comment (kik-kik) [inline sql/moz-fx-data-shared-prod/telemetry_derived/urlbar_clients_daily_v1/metadata.yaml:18]: Would it be worth specifying clustering fields?

Comment (jklukas) [reply sql/moz-fx-data-shared-prod/telemetry_derived/urlbar_clients_daily_v1/metadata.yaml:18]: Yes, this is likely an oversight. The table contains `normalized_channel`, which we usually do use as a clustering field when available.

File: sql/moz-fx-data-shared-prod/telemetry_derived/urlbar_clients_daily_v1/metadata.yaml [hunk 2]
```diff
@@ -1,12 +1,20 @@
...
 scheduling:
   dag_name: bqetl_urlbar
+bigquery:
+  time_partitioning:
+    field: submission_date
+    type: day
+    require_partition_filter: true
+  clustering:
+    fields:
+    - normalized_channel
```
Comment (jklukas) [inline sql/moz-fx-data-shared-prod/telemetry_derived/urlbar_clients_daily_v1/metadata.yaml:20]: I've enforced this new clustering by running:

```
bq update -clustering_fields=normalized_channel  moz-fx-data-shared-prod:telemetry_derived.urlbar_clients_daily_v1
```

File: sql/moz-fx-data-shared-prod/telemetry_derived/urlbar_clients_daily_v1/metadata.yaml (hunk 1 above)
Comment (kik-kik) [reply sql/moz-fx-data-shared-prod/telemetry_derived/urlbar_clients_daily_v1/metadata.yaml]: 🙇

**Pull request merged**